File: Exemplos M10/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/SAT/EnviarDadosVenda.ts
import {SatCommand} from './SatCommand';

export class EnviarDadosVenda extends SatCommand {
  readonly numSessao: number;
  readonly codAtivacao: string;
  readonly dadosVenda: string;

  constructor(numSessao: number, codAtivacao: string, dadosVenda: string) {
    super('EnviarDadosVenda');
    this.numSessao = numSessao;
    this.codAtivacao = codAtivacao;
    this.dadosVenda = dadosVenda;
  }

  functionParameters(): object {
    return {
      numSessao: this.numSessao,
      codAtivacao: this.codAtivacao,
      dadosVenda: this.dadosVenda,
    };
  }
}

File: Exemplos M10/App_IntentDigitalHUB_XamarinForms/Elgin_IntentDigitalHUB_XamarinForms-font/E1_Digital_Hub_Xamarin_Forms_M8/IntentServices/Sat/EnviarDadosVenda.cs
namespace Xamarin_Forms_Intent_Digital_Hub.IntentServices.Sat
{
    class EnviarDadosVenda : SatCommand
    {
        readonly int numSessao;
        readonly string codAtivacao;
        readonly string dadosVenda;

        public EnviarDadosVenda(int numSessao, string codAtivacao, string dadosVenda) : base("EnviarDadosVenda")
        {
            this.numSessao = numSessao;
            this.codAtivacao = codAtivacao;
            this.dadosVenda = dadosVenda;
        }

        protected override string FunctionParameters()
        {
            return "\"numSessao\"" + ":" + numSessao + "," +
                    "\"codAtivacao\"" + ":" + "\"" + codAtivacao + "\"" + "," +
                    "\"dadosVenda\"" + ":" + "\"" + dadosVenda + "\"";
        }
    }
}